### ISSUE TITLE
Improve scheduling of DOM updates

### ISSUE THREAD

**Issue by fabricedesre** [utkarshkukreti/draco#2]
It seems to me that Instance::render() doesn't happen in a requestAnimationFrame callback, which would spare extra work.

**Comment by utkarshkukreti** [comment]
I don't think that would be a good default. As far as I can see, none of the popular JS frameworks do that. Some of the downsides of doing it are in the comments [here](https://github.com/facebook/react/issues/11171).

I checked the source of a few frameworks and only Vue calls "requestAnimationFrame" and that's also only for scheduling animations, not the rendering.

https://unpkg.com/react@16.6.0/cjs/react.development.js
https://unpkg.com/vue@2.5.17/dist/vue.js
https://unpkg.com/inferno@6.0.3/dist/index.cjs.js
https://unpkg.com/preact@8.3.1/dist/preact.js

**Comment by fabricedesre** [comment]
Interesting threads, thanks! It still looks like having a facility to opt-in could be useful but that doesn't have to be part of the library.
